src/app/services/search.service.ts

```diff
@@ -0,0 +1,279 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { AppConfigService } from './app-config.service';
+import { fhirclient } from 'fhirclient/lib/types';
+import * as fhirpath from 'fhirpath';
+
+// Type alias for the patient bundle which presumably won't always be a string
+type PatientBundle = string;
+
+/**
+ * Marks a path.
+ */
+export type FHIRPath = string;
+
+export type ResearchStudy = fhirclient.FHIR.Resource;
+
+/**
+ * TrialScope facility. Will be removed eventually.
+ */
+export interface Facility {
+  facility: string;
+  contactPhone?: string;
+  contactEmail?: string;
+}
+
+/**
+ * Wrapper class for a research study. Provides hooks to deal with looking up
+ * fields that may be missing in the actual FHIR result.
+ */
+export class ResearchStudySearchEntry {
+  /**
+   * The embedded resource.
+   */
+  resource: fhirclient.FHIR.Resource;
+  private cachedSites: fhirpath.FHIRResource[] | null = null;
+  private containedResources: Map<string, fhirpath.FHIRResource> | null = null;
+
+  constructor(public entry: fhirclient.FHIR.BundleEntry) {
+    this.resource = this.entry.resource;
+    console.log(this.entry);
+  }
+
+  // These map existing TrialScope properties to their new FHIR types. They
```

Comment:
Update this comment to say something along the lines of: "These mappings are used for pulling out data to be exported to a csv"